src/main/java/org/moltimate/moltimatebackend/util/DockingUtils.java

```diff
@@ -72,4 +66,101 @@ public void transferTo(File file) throws IOException, IllegalStateException {
 			fos.write( bytes );
 		}
 	}
+
+	/**
+	 * Replace atoms in a protein (or ligand) file
+	 */
+	public static byte[] replaceAtoms(byte[] byteFile) throws IOException {
+		File sourcefile = new File("tmp");
+		FileUtils.writeByteArrayToFile(sourcefile, byteFile);
+    	String replaceTablePath = "";
+
+    	// NOTE: It is critical that named groups only exist for the atoms that we expect to replace
+		// Do not create named groups for anything other than the atoms subject to replacement.
+    	String pdbRegex =
+			"((ATOM)*(HETATM)*)\\s+" +
+			"((\\d+))\\s+" +
+			"(?<atom>(\\w+))\\s*" +
+			"(\\w{3})\\s+A\\s*\\d+\\s+" +
+			"((\\-?\\d+\\.\\d+))\\s+((\\-?\\d+\\.\\d+))\\s+((\\-?\\d+\\.\\d+))\\s+" +
+			"(\\d+\\.\\d+)\\s+((\\-?\\d+\\.\\d+))\\s+" +
+			"(?<atom2>(\\w+))\\s*";
+
+    	String cifRegex =
+			"(ATOM)\\s+\\d+\\s+" +
+			"(?<atom>\\w+)\\s*" +
+			"(?<atom2>\\w+)\\s+" +
+			"\\.\\s+\\w+\\s+\\w\\s+" +
+			"\\d+\\s+\\d+\\s+\\?\\s+" +
+			"\\-?\\d+\\.\\d+\\s+" +
+			"\\-?\\d+\\.\\d+\\s+" +
+			"\\-?\\d+\\.\\d+\\s+" +
+			"\\d+\\.\\d+\\s+" +
+			"\\-?\\d+\\.\\d+\\s+\\?\\s+\\d+\\s+\\w+\\s+\\w+\\s+" +
+			"(?<atom3>\\w+)\\s+\\d+\\s*";
+
+    	String sdfRegex =
+			"(\\-?\\d+\\.\\d+)\\s+" +
+			"(\\-?\\d+\\.\\d+)\\s+" +
+			"(\\-?\\d+\\.\\d+)\\s+" +
+			"(?<atom>\\w+)\\s+" +
+			"(\\-?\\d+\\.?\\d*)\\s+" +
+			"(\\-?\\d+\\.?\\d*)\\s+" +
+			"(\\-?\\d+\\.?\\d*)\\s+" +
+			"(\\-?\\d+\\.?\\d*)\\s+" +
+			"(\\-?\\d+\\.?\\d*)";
+
+    	// Select the regular expression that will be used to match lines based on the file's extension.
+		String workingRegex;
+		String filename = sourcefile.getName().toLowerCase();
+    	if( filename.endsWith(".pdb") ){
+			workingRegex = pdbRegex;
+		} else if(filename.endsWith(".cif")) {
+			workingRegex = cifRegex;
+		} else if(filename.endsWith(".sdf")){
+    		workingRegex = sdfRegex;
+		} else {
+    		throw new IllegalArgumentException("\'sourcefile\' parameter name does not contain a known filename");
+		}
+		Pattern linePattern = Pattern.compile(workingRegex);
+    	Matcher m;
+
+		// Create a temporary file in the current directory
+		File tempFile = File.createTempFile( "tmp", null, sourcefile.getAbsoluteFile().getParentFile() );
+		tempFile.createNewFile();
+
+		// TODO: Build this from a configuration file
+		Map<String, String> replaceTable = new HashMap<String, String>();
+		replaceTable.put("SE", "S");
```

Comment:
Are we adding the switches here for now before adding the config file?

Reply:
I was going to add them here for now, since I'm not sure how we'd go about exposing a configuration file on GCP. I can look into it more if we want it like that, though.